src/storage/transaction/ChainDeleteEdgesGroupProcessor.cpp

```diff
@@ -0,0 +1,75 @@
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License.
+ */
+
+#include "storage/transaction/ChainDeleteEdgesGroupProcessor.h"
+
+#include "storage/StorageFlags.h"
+#include "storage/transaction/ChainDeleteEdgesLocalProcessor.h"
+#include "storage/transaction/ConsistUtil.h"
+#include "storage/transaction/TransactionManager.h"
+
+namespace nebula {
+namespace storage {
+using ChainID = std::pair<PartitionID, PartitionID>;
+using SplitedRequest = std::unordered_map<ChainID, cpp2::DeleteEdgesRequest>;
+
+void ChainDeleteEdgesGroupProcessor::process(const cpp2::DeleteEdgesRequest& req) {
+  auto spaceId = req.get_space_id();
+  auto localPartId = req.get_parts().begin()->first;
+  auto stSplitRequest = splitRequest(req);
+  if (!stSplitRequest.ok()) {
+    // TODO(liuyu): change this when error code done
+    pushResultCode(Code::E_SPACE_NOT_FOUND, localPartId);
+    onFinished();
+  }
+
+  SplitedRequest splitedRequest = stSplitRequest.value();
+
+  callingNum_ = splitedRequest.size();
+
+  auto fnSplit = [&](auto& request) {
+    auto* proc = ChainDeleteEdgesLocalProcessor::instance(env_);
+    proc->getFuture().thenValue([=](auto&& resp) {
+      auto code = resp.get_result().get_failed_parts().empty()
+                      ? nebula::cpp2::ErrorCode::SUCCEEDED
+                      : resp.get_result().get_failed_parts().begin()->get_code();
+      handleAsync(spaceId, localPartId, code);
+    });
+    proc->process(request.second);
+  };
+
+  std::for_each(splitedRequest.begin(), splitedRequest.end(), fnSplit);
+}
+
+StatusOr<SplitedRequest> ChainDeleteEdgesGroupProcessor::splitRequest(
```

Comment:
Why not keep it consistent with the `shuffleRequest` in `ChainAddEdgesGroupProcessor`?

Reply:
I start to like StartOr ...